src/%IPM/Storage/Module.cls

```
@@ -512,25 +514,36 @@ Method LoadDependencies(pPhaseList As %List, ByRef pParams) As %Status
 	Quit tSC
 }
 
-Method BuildDependencyGraph(ByRef pDependencyGraph, pDepth As %Integer = 1, pForceSnapshotReload As %Boolean = 0, ByRef qstruct, pPhases As %String = "", ByRef pSubModules, pPass As %Integer = 1, pModulePath As %List = {$ListBuild(..Name)}, pIgnoreInstalledModules As %Boolean = 0, pKnownDependencies As %List = {..GetKnownDependencies(..Name)}, pPermitDowngrade As %Boolean = 0, pCheckNestedScoped As %Boolean = 0) As %Status
+/// DependencyGraphConfig.Graph:
+/// Graph("dependent module name") = $ListBuild(<depth>,<server name>,<exact version>,<module state>)
+/// Graph("dependent module name","<target module> <target module exact version>") = <dependent module semantic version>
+Method ConstructDependencyGraph(pGraphConfig As %IPM.Utils.DependencyGraphConfig) As %Status
```

Since we are refactoring, have it throw errors instead of returning status